lib/self-evict.js

```diff
@@ -179,10 +180,68 @@ SelfEvict.prototype._preEvict = function _preEvict() {
 };
 
 SelfEvict.prototype._evict = function _evict() {
+    var self = this;
+
     this._transitionTo(PhaseNames.Evicting);
 
-    this.ringpop.membership.setLocalStatus(Member.Status.faulty);
-    process.nextTick(this._postEvict.bind(this));
+    var pingEnabled = this.ringpop.config.get('selfEvictionPingEnabled');
+
+    var restartGossip = false;
+    // Stop gossip when currently gossiping and ping is enabled
+    if (!this.ringpop.gossip.isStopped && pingEnabled) {
```

Comment:
I think the code is a bit more readable when you remove the `pingEnabled` check here and move the complete block in the `pingEnabled` condition block below.